Update CHANGELOG.md

maxbeier opened the pull request:
Fixed typos and formatting. Linked issues and rules to their doc pages for faster reading.

ljharb commented on packages/eslint-config-airbnb/CHANGELOG.md:
```diff
@@ -1,126 +1,126 @@
 6.0.1 / 2016-02-21
 ==================
-- [fix] disable `newline-per-chained-call` due to an `eslint` bug (#748)
+- [fix] disable [`newline-per-chained-call`](http://eslint.org/docs/rules/newline-per-chained-call) due to an `eslint` bug [(#748)](https://github.com/airbnb/javascript/issues/748)
```
instead of typing out the whole URL, the issue number can be referenced with just `[#748](airbnb/javascript#748)` - let's do that.

For all the eslint rules, let's use named footnotes to link to them (the rule name is ideal) rather than typing the long URLs inline (and likely duplicating them as well)

maxbeier commented:
> instead of typing out the whole URL, the issue number can be referenced with just `[#748](airbnb/javascript#748)` - let's do that.

> For all the eslint rules, let's use named footnotes to link to them (the rule name is ideal) rather than typing the long URLs inline (and likely duplicating them as well)

Added something like footnotes and linked to them and shortened the links to issues.

ljharb commented:
Thanks - when I load up the markdown file in your branch, sadly my shortened links (which worked in my preview before I suggested it) don't seem to be working correctly, which implies that we do need the full links.

Let's restore the full issue/PR links, and rebase this down to one commit on latest master, and we'll get this merged!

maxbeier commented:
Hi,

sorry, I just learned that footnotes are supported in GFM and changed to them. I'm confused why the short links to the issues don't work (it seems this only works in issues itself (?)) and restored the absolute URLs.

Anyways, I rebased and hope it's better now.

ljharb commented on packages/eslint-config-airbnb/CHANGELOG.md:
```diff
@@ -1,132 +1,172 @@
 6.0.2 / 2016-02-22
 ==================
-- [fix] disable `no-confusing-arrow` due to an `eslint` bug (#752)
+- [fix] disable [`no-confusing-arrow`][no-confusing-arrow] due to an `eslint` bug ([#752](https://github.com/airbnb/javascript/issues/752))
```
curious tho, why remove the space before the `-`? In raw form I prefer how it looks with the space.

maxbeier replied:
Oh, no real reason, just for consistency – since 6.0.0 there was no space anymore and I thought that might be the new style, so I adjusted the rest.

ljharb replied:
gotcha, that wasn't intentional. nbd tho. thanks for the pr!

ljharb commented:
Looks great, thanks!
